tests/cohort_tests.py

```diff
@@ -0,0 +1,43 @@
+"""
+See
+https://github.com/NHSDigital/pytest-nhsd-apim/blob/main/tests/test_examples.py
+for more ideas on how to test the authorization of your API.
+"""
+import requests
+import pytest
+import json
+from lib import Generators
+from lib.constants import CORRELATION_IDS, URL_SLUG
+from lib.api_helpers import *
+
+expected_response_published_library_get_all = json.dumps(
+    json.load(published_cohort_library_get_all_expected_response), indent=1)
+
+
+@pytest.mark.functionaltest
+@pytest.mark.parametrize("correlation_id", CORRELATION_IDS)
+def test_for_getall_query(nhsd_apim_proxy_url, correlation_id):
+    published_cohort_definitions_response = requests.post(
+        f"{nhsd_apim_proxy_url}/api",
+        headers=Generators.generate_target_server_headers(correlation_id),
```

Comment:
its so can assert the correlation id has updated in response on each test

Reply:
Okay, I see - but bear in mind this is not the purpose of the test. From what I can we test `test_for_getall_query`, so I guess we're interested in getting all the cohorts, rather than updating a header.

That's why I asked a few lines below if there are separate tests for headers. We need a separation of concerns within our tests, and not to mix/overlap actual checks we make as part of our tests 👍  Happy to have a chat about it.